### DISCUSSION TITLE
E  File: http://localhost/ - Line 887 - Msg: TypeError: Cannot read properties of undefined (reading 'from')

### DISCUSSION THREAD

**Discussion by StefanZivkovic** [capawesome-team/capacitor-firebase#495] [Q&A] [answered]
i get this error in logs in android studio

``` 
Capacitor/...gingPlugin com...ingsoftware.bifrostSunsetAppD  V  Notifying listeners for event notificationReceived
2023-11-17 15:53:29.963  1860-5162  sendRespon...ssage ---- com..appId  {"save":true,"callbackId":"7530493",
"pluginId":"FirebaseMessaging", "methodName":"addListener","success":true,"data":{"notification":
{"id":"0:1700232809616178%134c6bca134c6bca",
"data":{"_title":"_title","_body":"_body"},"title":"title","body":"body",
"tag":"campaign_collapse_key_6219337983981968364"}}}
2023-11-17 15:53:29.966  1860-5162  Capacitor               com...appId  D  sendResponseMessage ---- 2
2023-11-17 15:53:29.996  1860-1860  Capacitor/Console       com...appId  E  File:
 http://localhost/ - Line 887 - Msg: TypeError: Cannot read properties of undefined (reading 'from')
 ```

This is when i send notification from firebase notification composer (firebase console, campaigns). I test the case when app is in background -  i need `notificationReceived` event to trigger in my javascript code when notification is displayed on android emulator

package: `@capacitor-firebase/messaging@5.2.0`, I also have `"@capacitor-community/fcm": "^5.0.2",`

I will continue to trace the log, but so far no success. Any thoughts?

**Comment by robingenz** [comment, marked as answer]
This is not a plugin error. This error is thrown by your web application. 

**Edit**: You should not use both packages together.

**Reply by StefanZivkovic** [reply]
oh, thanks!  Will look at it now!

From what i can see in the code, in that event handler (`notificationReceived`), there is such code 
```
if(notification.data.from && typeof notification.data.from === "string" && notification.data.from.includes("topic")) 
```
which was there, and before I have used official plugin `@capacitor/push-notifications`. Maybe there are some differences how these 2 packages prepare message/notification data, hm?

For now, i will guard on `notification.data` just to see what happens

**EDIT**: yes, i removed `@capacitor/push-notifications` previously. So i have `@capacitor-firebase/messaging` and yes, i still have `@capacitor-community/fcm`. Is that of big issue? i see `@capacitor-firebase/messaging` contains all those tokens/subscriptions methods, but at the moment i don't need it so i did not want to bother to make more changes. But yes, if deployed (to prod), it should pick one of those.

Seems like the structure is `notification.notification.{...., data, ...rest}`, which causes issues. OK, but pointing me to javascript was of  a key help here. Thanks!

**Reply by robingenz** [reply]
`@capacitor-firebase/messaging` replaces both packages, `@capacitor/push-notifications` and `@capacitor-community/fcm`. You can try to use both packages but this is not officially supported.
